### PR TITLE
docs(examples): update gauge example

### PR DESCRIPTION
**Pull request by Emivvvvv**
Issue: https://github.com/ratatui-org/ratatui/issues/846

This PR adds:

for gauge.rs

- subjectively better-looking front
- impl `Widget` for `&mut App`

for gauge.tape

- change to get better output from the new code

Here is the new output:

gauge

### CODE REVIEW

File: examples/gauge.rs [hunk 1]
```diff
@@ -32,36 +36,54 @@ struct AppState {
...
+    let backend = CrosstermBackend::new(stdout());
+    let terminal = Terminal::new(backend)?;
+    Ok(terminal)
+}
+
+fn restore_terminal() -> color_eyre::Result<()> {
+    disable_raw_mode()?;
+    stdout().execute(LeaveAlternateScreen)?;
+    Ok(())
+}
```
Comment (Valentin271) [inline examples/gauge.rs:79]: You may just move those to the end of the file so we can better see the actual example code.

**Review by Valentin271 — APPROVED** [review]
LGTM, code and style wise.

File: examples/gauge.rs (hunk 1 above)
Comment (Emivvvvv) [reply examples/gauge.rs:79]: Yeah makes sense

File: examples/gauge.tape [hunk 2]
```diff
@@ -3,10 +3,12 @@
...
-Set Height 550
+Set Height 850
 Hide
 Type "cargo run --example=gauge --features=crossterm"
 Enter
-Sleep 1s
+Sleep 2s
 Show
+Sleep 2s
+Enter 1
```
Comment (joshka) [inline examples/gauge.tape:13]: Great idea!

File: examples/gauge.rs [hunk 3]
```diff
@@ -70,105 +63,161 @@ impl App {
...
-            let state = self.state;
-            let layout = Layout::vertical([Constraint::Ratio(1, 4); 4]).split(frame.size());
-            Self::render_gauge1(state.progress1, frame, layout[0]);
-            Self::render_gauge2(state.progress2, frame, layout[1]);
-            Self::render_gauge3(state.progress3, frame, layout[2]);
-            Self::render_gauge4(state.progress4, frame, layout[3]);
-        })?;
-        Ok(())
-    }
+impl App {
```
Comment (joshka) [inline examples/gauge.rs:68]: Can be joined with the above impl block.

File: examples/gauge.rs [hunk 4]
```diff
@@ -32,36 +36,25 @@ struct AppState {
...
+    Ok(())
+}
+
+impl App {
+    fn new() -> App {
+        App {
+            state: AppState::default(),
+            start: false,
+        }
     }
```
Comment (joshka) [inline examples/gauge.rs:58]: Can be replaced with derive default

**Review by joshka — APPROVED** [review]
This looks really good. My comments are purely about simplifying things as much as possible for readers.

File: examples/gauge.rs [hunk 5]
```diff
@@ -70,105 +63,161 @@ impl App {
...
+            .gauge_style(GAUGE4_COLOR)
             .percent(progress)
-            .label(label);
-        frame.render_widget(gauge, area);
+            .label(label)
+            .render(area, buf);
+    }
+
+    fn render_footer(&self, area: Rect, buf: &mut Buffer) {
+        Paragraph::new("Press ENTER to start")
```
Comment (joshka) [inline examples/gauge.rs:179]: Add 'q' to quit here.

**Comment by joshka** [comment]
Made with VHS

**Pull request merged**